edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/ImagePullSecretName.cs

```diff
@@ -8,17 +8,37 @@ namespace Microsoft.Azure.Devices.Edge.Agent.Kubernetes.EdgeDeployment
     [JsonConverter(typeof(ImagePullSecretNameConverter))]
     public class ImagePullSecretName
     {
+        const string HttpStart = "http://";
+        const string HttpsStart = "https://";
+
         readonly string value;
 
         public ImagePullSecretName(string name)
         {
             this.value = Preconditions.CheckNonWhiteSpace(name, nameof(name));
         }
 
-        public static ImagePullSecretName Create(AuthConfig auth) => new ImagePullSecretName($"{auth.Username.ToLower()}-{auth.ServerAddress.ToLower()}");
+        public static ImagePullSecretName Create(AuthConfig auth) => new ImagePullSecretName(KubeUtils.SanitizeDNSDomain($"{auth.Username.ToLower()}-{NamifyServerAddress(auth.ServerAddress)}"));
 
         public static implicit operator string(ImagePullSecretName name) => name.ToString();
 
+        // A variation on Moby project's ConvertToHostname
+        private static string NamifyServerAddress(string serverAddress)
+        {
+            string stripped = serverAddress.ToLower();
+            if (serverAddress.StartsWith(HttpsStart))
+            {
+                stripped = stripped.Substring(HttpsStart.Length);
+            }
+            else if (serverAddress.StartsWith(HttpStart))
+            {
+                stripped = stripped.Substring(HttpStart.Length);
+            }
+
+            string[] nameparts = stripped.Split('/', 2);
+            return nameparts[0].Replace(':', '-');
+        }
```

Comment:
Is it more standard to use uri.host instead of coding one? https://docs.microsoft.com/en-us/dotnet/api/system.uri?view=netframework-4.8

Reply:
You know what stinks?  Registry addresses aren't URIs.  I did attempt to use the URI class here, eventually, I had to settled on implementing `ConvertToHostName` from the Moby project.

Reply:
re: "Registry addresses aren't URIs". That's a bit weird because if I put all the examples you have - in ValidServerAddresses method - in a URI validator, the are actually valid URIs. But since you tried that URI class and it did not work, it lgtm.

Reply:
Using the URI class failed on this test: "my-acr.io:8080" the URI class interprets that as protocol="my-acr.io" with hostname="8080" :(